Source/ACE.Server/Factories/LootGenerationFactory.cs

```diff
@@ -1545,78 +1488,108 @@ private static int GetWorkmanship(int tier)
-                    break;
-                case 3:
-                    spellcraft = ThreadSafeRandom.Next(70, 90) + spellAmount * ThreadSafeRandom.Next(1, 6); //80 - 130
-                    break;
-                case 4:
-                    spellcraft = ThreadSafeRandom.Next(100, 120) + spellAmount * ThreadSafeRandom.Next(1, 7); /// 120 - 160
-                    break;
-                case 5:
-                    spellcraft = ThreadSafeRandom.Next(130, 150) + spellAmount * ThreadSafeRandom.Next(1, 8); ///150 - 210
-                    break;
-                case 6:
-                    spellcraft = ThreadSafeRandom.Next(160, 180) + spellAmount * ThreadSafeRandom.Next(1, 9); /// 200-260
-                    break;
-                case 7:
-                    spellcraft = ThreadSafeRandom.Next(230, 260) + spellAmount * ThreadSafeRandom.Next(1, 10); /// 250 - 310
-                    break;
-                case 8:
-                    spellcraft = ThreadSafeRandom.Next(280, 300) + spellAmount * ThreadSafeRandom.Next(1, 11); //300-450
-                    break;
+                case ItemType.MeleeWeapon:
+                case ItemType.Caster:
+                case ItemType.MissileWeapon:
+                case ItemType.Armor:
+                case ItemType.Clothing:
+                case ItemType.Jewelry:
+                    minItemSpellCraftRange = 0.90f;
+                    maxItemSpellCraftRange = 1.05f;
+                    break;
+                case ItemType.Gem:
                 default:
+                    minItemSpellCraftRange = 1.00f;
+                    maxItemSpellCraftRange = 1.00f;
                     break;
             }
 
-            return spellcraft;
+            // Getting the spell difficulty - Maybe a better way to do this.
+            var maxSpellLevel = wo.GetMaxSpellLevel();
+            int maxSpellDiff = 1;
+
+            if (maxSpellLevel == 2)
+                maxSpellDiff = 50;
+            else if (maxSpellLevel == 3)
+                maxSpellDiff = 100;
+            else if (maxSpellLevel == 4)
+                maxSpellDiff = 150;
+            else if (maxSpellLevel == 5)
+                maxSpellDiff = 200;
+            else if (maxSpellLevel == 6)
+                maxSpellDiff = 250;
+            else if (maxSpellLevel == 7)
+                maxSpellDiff = 300;
+            else if (maxSpellLevel == 8)
+                maxSpellDiff = 400;  
+
+            var tItemSpellCraft = maxSpellDiff * ThreadSafeRandom.Next(minItemSpellCraftRange, maxItemSpellCraftRange);
+
+            if (tItemSpellCraft < 0)
+                tItemSpellCraft = 0;
+
+            int finalItemSpellCraft = (int)Math.Floor(tItemSpellCraft);
+
+            return finalItemSpellCraft;
+
         }
 
-        private static int GetDifficulty(int tier, int spellcraft)
+        private static int GetDifficulty(WorldObject wo, int tier, int itemspellcraft)
```

Comment:
is tier no longer a requirement for this function?

i see that tier is still passed in as a parameter, but it seems to no longer be used